### ISSUE TITLE
RTSP not working

### ISSUE THREAD

**Issue by gusarg81** [ghoost82/mijia-720p-hack#22]
Hi,

   Today test the last from git and what does not work:

- RTSP does not work (daemon crash). When tried by hand, it shows:

> ioctl "VPD_GET_PLATFORM_VERSION" fail: Invalid argument
[err] gm_init:
        Platform is not ready!, please check inser module driver flow.
ioctl "VPD_ENV_UPDATE" driver into stop state.

- mijia_ctrl does not work:

> mijia_ctrl 
Failed to open /dev/ftpwmtmr010
PWM initialization failed
UDP server started on 0.0.0.0:8888
Exit with Ctrl+C

Any Ideas?

Thanks in advance.

**Comment by bt4wang** [comment]
I've already downgrade fw to 3.3.6_2017071114, but rtsp doesn't work either. I can see the http page, but the rtsp link has no respose. Don't know what to do. It seems that noone is maintaining this project any more.

**Comment by gusarg81** [comment]
Yeah, seems dead. A Shame...

**Comment by SuperSpe** [comment]
same problem here
# /tmp/sd/mijia-720p-hack/bin/rtspd -m
  Usage:
  Multiple streams:
    #./rtspd
  Max single h264 stream:
    #./rtspd -s
  Max single mpeg4 stream:
    #./rtspd -m
  Max single mjpeg stream:
    #./rtspd -j
Error! Abnormal AP exit under chipid(0),cpuid(1),graph(1),query_id=0x10001
ioctl "VPD_APPLY" fail! (ret=-1)
[err]   gm_apply groupfd(0x1):
        Apply fail!
ioctl "VPD_POLL" driver into stop state.


anyone managed to solve?

**Comment by vkizhaev-appfire** [comment]
I was experiencing the very same problems, but finally got it resolved.

What helped me was to re-do the whole process from scratch:
1. Re-install the firmware (3.3.6_2017092810)
2. Connect camera to MiHome (seems like an essential step)
3. Install the hack

**Comment by gusarg81** [comment]
@vkizhaev-spartez can you send me that firmware please? I can' t find it.

Thanks.

**Comment by pekidi** [comment]
> @vkizhaev-spartez can you send me that firmware please? I can' t find it.
> 
> Thanks.

hi,
recovery image is possible to get here
https://c720p-recovery.pekidi.com
Use tf_recovery.img file and report

**Comment by vkizhaev-appfire** [comment]
[tf_recovery (1).img.zip](https://github.com/ghoost82/mijia-720p-hack/files/5628296/tf_recovery.1.img.zip)
This is the one I used. Though I must say, that the whole thing didn't worth it in my case, because I ended up having an RTSP stream with a lag of about 10 seconds and the picture in MiHome had become a mess, full of broken squares upon each movement and such stuff. I'm using the camera as a baby cam and as you can imagine, my wife didn't like the change :) 

Moreover, my camera can't rotate now from the MiHome app, only manually, but I don't know if it has anything to do with the hack (removing the card doesn't help).

**Comment by gusarg81** [comment]
Yeah, I've read about the poor RTSP. So, basically is a useless camera (I was not using it anyways). Every camera with no RTSP and ONVIF is kind useless...

Also, I can't even make it work normally, does not reads the QR code in Xiaomi Home app when trying to add the device.
